haystack/core/pipeline/pipeline.py

```diff
@@ -379,13 +379,23 @@ def run(  # noqa: PLR0915, PLR0912, C901, pylint: disable=too-many-branches
                                 pipeline_snapshot=new_pipeline_snapshot, pipeline_outputs=pipeline_outputs
                             )
 
-                component_outputs = self._run_component(
-                    component_name=component_name,
-                    component=component,
-                    inputs=component_inputs,  # the inputs to the current component
-                    component_visits=component_visits,
-                    parent_span=span,
-                )
+                try:
+                    component_outputs = self._run_component(
+                        component_name=component_name,
+                        component=component,
+                        inputs=component_inputs,  # the inputs to the current component
+                        component_visits=component_visits,
+                        parent_span=span,
+                    )
+                except Exception as e:
+                    if isinstance(e, BreakpointException):
```

Comment:
Could we also add this mechanism to `AsyncPipeline`? I'm sure people would appreciate having this there as well.